tests-read-only/oa/oa-css.spec.js

```diff
@@ -1,34 +1,75 @@
-const fs = require('fs');
-const { JSDOM } = require('jsdom');
-const css = require('css');
-
+/**
+ * @jest-environment jsdom
+*/
+import fs from 'fs';
+import css from 'css';
+import { renderView } from '../../src/data.js';
 const html = fs.readFileSync('./src/index.html', 'utf-8');
-const page = new JSDOM(html);
-const { window } = page;
-const { document } = window;
+document.body.innerHTML = html;
 
 const stylesPath = document.querySelector('link[rel="stylesheet"]').getAttribute('href');
 const style = fs.readFileSync('./src/' + stylesPath, 'utf-8');
 const { rules } = css.parse(style).stylesheet;
 
 const BOX_MODEL_ATTRIBUTES = ['width', 'height', 'margin', 'padding', 'border', 'box-sizing', 'background'];
-
+const FLEXBOX_ATTRIBUTES = ['flex-wrap', 'flex-direction'];
+// , flex-direction, justify-, align-
 const getRulesForSelector = (selector) => {
   return rules.filter(
     (rule) =>
       rule.type === 'rule' &&
-      rule.selectors.some((s) => s.trim() === selector)
+      rule.selectors.some((s) => s.trim().includes(selector))
   );
 }
 
-describe('CSS', () => {
-
-  const select = document.querySelector('select');
-  const selectClasses = Array.from(select.classList.values());
+const fakeData = [
+  {
+    name: "charizard",
+    img: "https://www.serebii.net/pokemongo/pokemon/006.png",
+    num: '006',
+    type: [
+      "fire",
+      "flying"
+    ],
+  },
+  {
+    name: "charmeleon",
+    img: "https://www.serebii.net/pokemongo/pokemon/005.png",
+    num: '005',
+    type: [
+      "fire"
+    ],
+  },
+];
+document.querySelector('#root').innerHTML = renderView(fakeData);
 
-  // const lis = Array.from(ul.querySelectorAll('li'));
+describe('CSS', () => {
+  const ul = document.querySelector('#root ul');
+  const cardsLi = ul.querySelectorAll('li');
+  const ulClasses = Array.from(ul.classList.values());
 
   describe('Uso de selectores de CSS', () => {
+    it('li elementos tienen class', () => {
+      cardsLi.forEach(item=>{
+        expect(item.getAttribute('class')).toBe('card');
+      });
+    });
+
+    it('Uso de flexbox', () => {
+      let allRulesAttributes = [];
+      ulClasses.forEach((ulClass) => {
+        const ulRules = getRulesForSelector(`.${ulClass}`);
+        const ulRulesAttributes = ulRules[0].declarations.map((declaration) => declaration.property);
+        allRulesAttributes = allRulesAttributes.concat(ulRulesAttributes);
+      });
```

Comment:
Prueba si esta funciona
```suggestion
      const allRulesAttributes = ulClasses.reduce((allRules, ulClass) => {
        const rules = getRulesForSelector(`.${ulClass}`);
        const ulRulesAttributes = rules[0].declarations.map((declaration) => declaration.property);
        return [...allRules, ulRulesAttributes];
      }, []);
```

Reply:
Funciona, cambie la ultima parte y use un concat igual, para evitar el array de arrays que salia.